src/mapclient/view/dialogs/selfclosing/messagebox.py

```diff
@@ -4,6 +4,21 @@
 
 class MessageBox(QtWidgets.QMessageBox):
     def __init__(self, icon, title, text, buttons, parent=None, close_after=None):
+        """
+        An extension of the Qt.QMessageBox class that automatically closes itself after a set number of seconds. This class allows the user
+        to pass an additional argument (close_after) to specify how long the MessageBox should be displayed for before it terminates.
+
+        Parameters
+        ----------
+        icon: PySide2.QtWidgets.QMessageBox.Icon
+        title: str
+        text: str
+        buttons: PySide2.QtWidgets.QMessageBox.StandardButtons
+        parent: PySide2.QtWidgets.QWidget, optional
+        close_after: float, optional
+            Specifies the number of seconds that the MessageBox should be displayed for before it closes. If the close_after argument is
+            set to zero, the MessageBox will not be displayed.
```

Comment:
I think I would put this in the main part of the documentation and just leave the Parameters section as types.

Reply:
Sweet. Done.